Test failure when using tornado 4.4

Can't have a 1.0 release without a new bug! Thankfully, it's an edge case caught by our tests.

```
======================================================================
FAIL: A response cell should allow the response status code to be set
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/conda/lib/python3.5/site-packages/tornado/testing.py", line 136, in __call__
    result = self.orig_method(*args, **kwargs)
  File "/opt/conda/lib/python3.5/site-packages/tornado/testing.py", line 529, in post_coroutine
    timeout=timeout)
  File "/opt/conda/lib/python3.5/site-packages/tornado/ioloop.py", line 457, in run_sync
    return future_cell[0].result()
  File "/opt/conda/lib/python3.5/site-packages/tornado/concurrent.py", line 237, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 3, in raise_exc_info
  File "/opt/conda/lib/python3.5/site-packages/tornado/gen.py", line 1024, in run
    yielded = self.gen.send(value)
  File "/opt/conda/lib/python3.5/types.py", line 177, in send
    return self.__wrapped.send(val)
  File "/srv/kernel_gateway/kernel_gateway/tests/test_notebook_http.py", line 290, in test_setting_response_status_code
    self.assertEqual(response.code, 204, 'Response status was not 204')
AssertionError: 500 != 204 : Response status was not 204
```

```
[E 160715 19:50:51 concurrent:341] Future <tornado.concurrent.Future object at 0x7f377457dcf8> exception was never retrieved: Traceback (most recent call last):
      File "/opt/conda/lib/python3.5/site-packages/tornado/gen.py", line 1024, in run
        yielded = self.gen.send(value)
      File "/srv/kernel_gateway/kernel_gateway/notebook_http/handlers.py", line 227, in _handle_request
        self.finish()
      File "/opt/conda/lib/python3.5/site-packages/tornado/web.py", line 939, in finish
        assert not self._write_buffer, "Cannot send body with %s" % self._status_code
    AssertionError: Cannot send body with 204
```

--- a/kernel_gateway/notebook_http/handlers.py
+++ b/kernel_gateway/notebook_http/handlers.py
@@ -188,10 +188,8 @@ class NotebookAPIHandler(TokenAuthorizationMixin,
 
             # Run the request and source code and yield until there's a result
             access_log.debug('Request code for notebook cell is: {}'.format(request_code))
-            request_future = self.execute_code(kernel_client, kernel_id, request_code)
-            yield request_future
-            source_future = self.execute_code(kernel_client, kernel_id, source_code)
-            source_result = yield source_future
+            yield self.execute_code(kernel_client, kernel_id, request_code)
+            source_result = yield self.execute_code(kernel_client, kernel_id, source_code)
 
             # If a response code cell exists, execute it
             if self.request.method in self.response_sources:
@@ -212,7 +210,8 @@ class NotebookAPIHandler(TokenAuthorizationMixin,
                     self.set_status(response['status'])
 
             # Write the result of the source code execution
-            self.write(source_result)
+            if source_result:
+                self.write(source_result)
         # If there was a problem executing an code, return a 500
         except CodeExecutionError as err:
             self.write(str(err))

